### PR TITLE
Add coverage job

### LINKED ISSUE #555: Add lcov coverage report to CI jobs
There are various error cases and stray blocks of code that appear not to be covered by the unit tests.

We should do a little bit more monitoring of the `lcov` report, and close any loopholes.

Noticed while checking for missed lines or branches in the new extended classes code.

### PR DESCRIPTION
**Pull request by NWilson**
Fixes #555

### CODE REVIEW

**Comment by zherczeg** [comment]
This looks useful.

`pcre2_jit_compile.c.html#L102: 58.9k #define MACHINE_STACK_SIZE 32768`

Interesting that a const has coverage. Some macros has coverage, others don't, so probably not all code is fully "covered" by the tool :)

**Comment by NWilson** [comment]
I'd guess that constants which are built into the binary will be counted as "code" and generate coverage.

Macros which are only used by the preprocessor won't generate coverage. They have to actually write out some text which is seen as C code by the compiler.

I'd be interested in trying out the "Coveralls" or "CodeCov" services, which look like good ways to track coverage over time, coverage diff in a PR, and so on.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -129,3 +129,61 @@ jobs:
...
+            ../src/ ./ \
+            >> "$GITHUB_STEP_SUMMARY"
+          echo '```' >> "$GITHUB_STEP_SUMMARY"
+
+      - name: Upload report
+        uses: actions/upload-artifact@v4
+        with:
+          name: "Coverage report"
+          path: './build/coverage-report'
+          if-no-files-found: error
```
Comment (carenas) [inline .github/workflows/build.yml:189]: missing "\n" at EOF

**Review by carenas — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (NWilson) [reply .github/workflows/build.yml]: Thanks! Fixed

**Review by carenas — APPROVED** [review]
it might be better to move this job into their own "tools.yml" in te long run, though

**Comment by NWilson** [comment]
> it might be better to move this job into their own "tools.yml" in te long run, though

I feel like code-scanning tools (like CodeQL, or Clang Static Analyzer) should be in their own jobs.

However, code coverage should be part of the Release "Build" pipeline (not the "Dev" pipeline) because it's actually a fixed, reportable artifact of the release process: "we built the code; we ran the tests; the tests covered 85% of the code; now we release the binaries".

**Pull request merged**